client/api/notebook.py

```diff
@@ -21,10 +21,7 @@ def run(self, protocol, messages, **kwargs):
         return self.assignment.protocol_map[protocol].run(messages, **kwargs)
 
     def auth(self, force=False, inline=True):
-        if not inline:
-            ok_auth.authenticate(self.assignment, force=force)
-        else:
-            ok_auth.notebook_authenticate(self.assignment, force=force)
+        self.assignment.authenticate(force=force, inline=inline)
```

Comment:
I put that logic into `self.assignment.authenticate`. Not sure if that's the right place it should go.